### PR TITLE
Prepare for 1.24.0 release

### LINKED ISSUE #44: Last published version of io.opentelemetry.semconv:opentelemetry-semconv
Hello,

I am following what's described in [opentelemetry-java 1.31.0 release notes](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.31.0):

> BREAKING: Stop publishing `io.opentelemetry:opentelemetry-semconv`. Please use `io.opentelemetry.semconv:opentelemetry-semconv:1.21.0-alpha` instead, which is published from [open-telemetry/semantic-conventions-java](https://github.com/open-telemetry/semantic-conventions-java). The new repository is published in lockstep with [open-telemetry/semantic-conventions](https://github.com/open-telemetry/semantic-conventions). (https://github.com/open-telemetry/opentelemetry-java/pull/5807)

When I look at the [releases of open-telemetry/semantic-conventions](https://github.com/open-telemetry/semantic-conventions/releases), the last release is `1.24.0`, and yet, when I go to [maven central](https://central.sonatype.com/artifact/io.opentelemetry.semconv/opentelemetry-semconv), the only release I can find is `1.23.1-alpha`. 

Since it is supposed to be "published in lockstep", I was expecting to find `1.24.0` in maven central.

Did I misunderstand something or is the publication mechanism broken?

Thanks

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #44.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,39 @@
...
+
+* BREAKING: Split out incubating artifact. This repo now
+  publishes `io.opentelemetry.semconv:opentelemetry-semconv:{{version}}` for semantic conventions
+  which are stable (the artifact itself is not yet stable but we aim to stabilize eventually),
+  and `io.opentelemetry.semconv:opentelemetry-semconv-incubating:{{version}}` for semantic
+  conventions which are incubating (experimental) (this artifact will always have the `-alpha`
+  designation). As a part of this restructuring, old attributes which do not exist
+  in [semantic-conventions](https://github.com/open-telemetry/semantic-conventions), but which we
+  had retained manually with deprecated annotations, have been removed.
+  ([#41](https://github.com/open-telemetry/semantic-conventions-java/pull/41))
```
Comment (trask) [inline CHANGELOG.md:18]: I think there's only a few of these?
```suggestion
  designation). As a part of this restructuring, a few old attributes which were removed from
  [semantic-conventions](https://github.com/open-telemetry/semantic-conventions) (without being deprecated there) have been removed.
  ([#41](https://github.com/open-telemetry/semantic-conventions-java/pull/41))
```

**Review by trask — APPROVED** [review]
🎈

File: CHANGELOG.md (hunk 1 above)
Comment (jack-berg) [reply CHANGELOG.md:18]: Here's the full list: https://github.com/open-telemetry/semantic-conventions-java/blob/1f90bc5dc388fb6ee13e179534098bbd5fe63fab/buildscripts/templates/SemanticAttributes.java.j2#L170-L941

If you do a search for `@Deprecated`, you get 74 results. For an accurate answer, you'd need to do an analysis to subtract the attributes which still remain but in a different class. So its somewhere between 0 and 74, and I'd guess that its closer to 74 than 0 considering that we only recently starting keeping things around in semconv.

**Review by breedx-splk — APPROVED** [review]
🌶️ Spicy, I love it.

**Pull request merged**